cmake/pika_setup_cuda.cmake

```diff
@@ -92,9 +91,18 @@ if(PIKA_WITH_CUDA AND NOT TARGET Cuda::cuda)
       INTERFACE $<$<COMPILE_LANGUAGE:CUDA>: --extended-lambda --default-stream
                 per-thread --expt-relaxed-constexpr >
     )
-  else()
     if(NOT PIKA_FIND_PACKAGE)
-      pika_add_target_compile_option(-DBOOST_THREAD_USES_MOVE PUBLIC)
+      if(PIKA_WITH_COMPILER_WARNINGS_AS_ERRORS)
+        target_compile_options(
+          pika_private_flags INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--Werror
+                                       all-warnings>
+        )
+      endif()
+
+      target_compile_options(
+        pika_private_flags
+        INTERFACE $<$<COMPILE_LANGUAGE:CUDA>:--display-error-number>
```

Comment:
This is a flyby that I've added. It means that an error code will be printed together with warnings/errors and in the future we can use those codes to suppress warnings locally.